cellxgene_schema_cli/cellxgene_schema/atac_seq.py

```diff
@@ -0,0 +1,384 @@
+    if not is_primary_data.all():
+        return "Anndata.obs.is_primary_data must all be True."
+
+
+def validate_anndata_feature_reference(anndata_file: str) -> Optional[str]:
+    with h5py.File(anndata_file) as f:
+        unique_feature_reference = ad.io.read_elem(f["var"])["feature_reference"].unique()
+        unique_organism_ontology_term_ids = ad.io.read_elem(f["obs"])["organism_ontology_term_id"].unique()
+    allowed_terms = [*feature_reference_by_chromosome_length_table.keys()]
+    if unique_feature_reference[0] not in allowed_terms:
+        return f"Anndata.var.feature_reference must be one of {allowed_terms}."
+    if unique_organism_ontology_term_ids != unique_feature_reference:
+        return "Unique Anndata.obs.organism_ontology_term_id must be equal to unqiue Anndata.var.feature_reference."
+
+
+def detect_chromosomes(parquet_file: str) -> list[str]:
+    logger.info("detecting chromosomes")
+    df = ddf.read_parquet(parquet_file, columns=["chromosome"]).drop_duplicates()
+    return df["chromosome"].values.compute()
+
+
+def index_fragment(fragment_file: str, parquet_file: str, tempdir: str, override_write_algorithm: Optional[str] = None):
+    # sort the fragment by chromosome, start coordinate, and stop coordinate, then compress it with bgzip
+    bgzip_output_file = fragment_file.replace(".gz", ".bgz")
+    bgzip_output_path = Path(bgzip_output_file)
+    bgzip_output_path.unlink(missing_ok=True)
+    bgzip_output_path.touch()
+    bgzip_write_lock = dd.Lock()  # lock to preserver write order
+
+    if override_write_algorithm:
+        write_algorithm = write_algorithm_by_callable[override_write_algorithm]
+    elif not shutil.which("bgzip"):  # check if bgzip cli is installed
+        logger.warning("bgzip is not installed, using slower pysam implementation")
+        write_algorithm = write_algorithm_by_callable["pysam"]
+    else:
+        write_algorithm = write_algorithm_by_callable["cli"]
+
+    chromosomes = detect_chromosomes(parquet_file)
+    jobs = prepare_fragment(chromosomes, parquet_file, bgzip_output_file, tempdir, bgzip_write_lock, write_algorithm)
+    # limit calls to dask.compute to improve performance. The number of jobs to run at once is determined by the
+    # step variable. If we run all the jobs in the same call to dask.compute, the local cluster hangs.
+    # TODO: investigate why
+    step = 4
+    # print the progress of the jobs
+    for i in range(0, len(jobs), step):
+        dask.compute(jobs[i : i + step])
+
+    logger.info(f"Fragment sorted and compressed: {bgzip_output_file}")
+    #
+    pysam.tabix_index(bgzip_output_file, preset="bed", force=True)
+    tabix_output_file = bgzip_output_file + ".tbi"
+    logger.info(f"Index file generated: {tabix_output_file}")
+
+
+@delayed
+def sort_fragment(parquet_file: str, write_path: str, chromosome: str) -> Path:
+    temp_data = Path(write_path) / f"temp_{chromosome}.tsv"
```

Comment:
```suggestion
    temp_data = Path(write_path) / f"temp_{chromosome}.tsv.gz"
```

I think this file should be compressed to keep storage requirements during processing down.

Ideally, I would use a better compressor than GZIP (like zstd) but I'm not sure what you'll have installed here.